src/composables/draw/draw-utils.ts

```diff
@@ -0,0 +1,294 @@
+import OlMap from 'ol/Map'
+import LineString from 'ol/geom/LineString.js'
+import MultiPoint from 'ol/geom/MultiPoint.js'
+import Polygon from 'ol/geom/Polygon.js'
+import Point from 'ol/geom/Point.js'
+import StyleCircle, { Options as CircleOptions } from 'ol/style/Circle.js'
+import StyleFill from 'ol/style/Fill.js'
+import StyleIcon from 'ol/style/Icon.js'
+import StyleRegularShape, {
+  Options as RegularShapeOptions,
+} from 'ol/style/RegularShape.js'
+import StyleText from 'ol/style/Text.js'
+import StyleStroke from 'ol/style/Stroke.js'
+import StyleStyle, { StyleLike } from 'ol/style/Style.js'
+import { Feature } from 'ol'
+// TODO 3D
+// import { transform } from 'ol/proj'
+// import olcsCore from 'olcs/core.js';
+
+const MYMAPS_URL = import.meta.env.VITE_MYMAPS_URL
+const MYMAPS_SYMBOL_URL = MYMAPS_URL + '/symbol/'
+const ARROW_URL = MYMAPS_URL + '/getarrow'
+// TODO 3D
+// const ARROW_MODEL_URL = import.meta.env.VITE_ARROW_MODEL_URL
+
+export function createStyleFunction(curMap: OlMap) {
```

Comment:
We should rework this, seperate default style, style for point and style for label, style for selectiion, etc... the v3 code is not very clear, it is hard to tell what style is for.

Reply:
partly done,
common state  of type DrawnFeature[]